File: TestDome/src/testdome/java/Song.java
package testdome.java;

import java.util.HashSet;

public class Song {
    private String name;
    private Song nextSong;

    public Song(String name) {
        this.name = name;
    }

    public void setNextSong(Song nextSong) {
        this.nextSong = nextSong;
    }

    public boolean isInRepeatingPlaylist() {
        boolean isRepeat = false;


//        if(nextSong.nextSong != null){
//            return true;
//        }

    //    LinkedList<Song> playlist = new LinkedList<>();

        //for speed use HashSet, hashSet is generally faster compared to LinkedList
        HashSet<Song> playlist = new HashSet<>();

        Song song = nextSong;

        while(song !=null){
            if(playlist.contains(song)){
                return true;
            }
            playlist.add(song);
            song = song.nextSong;
        }


        return isRepeat;
    }

    public static void main(String[] args) {
        Song first = new Song("Hello");
        Song second = new Song("Eye of the tiger");

        first.setNextSong(second);
        second.setNextSong(first);

        System.out.println(first.isInRepeatingPlaylist());
    }
}

File: TestDome/src/testdome/javascript/Song.js
class Song {
  name;
  nextSong;

  constructor(name) {
    this.name = name;
  }

  /**
   * @return {boolean} true if the playlist is repeating, false if not.
   */
  isInRepeatingPlaylist() {
    // Your code goes here
    let isRepeat = false;
    const playlist = new Set();

    let song = this.nextSong;


        while(song !=null){
            if(playlist.has(song)){
                return true;
            }
            playlist.add(song);
            song = song.nextSong;
        }


        return isRepeat;


  }
}

let first = new Song("Hello");
let second = new Song("Eye of the tiger");

first.nextSong = second;
second.nextSong = first;

console.log(first.isInRepeatingPlaylist());